recipe/meta.yaml

```diff
@@ -89,7 +90,7 @@ outputs:
         - cvxpy.problems
         - cvxpy.reductions
         - cvxpy.reductions.complex2real
-        - cvxpy.reductions.complex2real.atom_canonicalizers
+        - cvxpy.reductions.complex2real.canonicalizers
         - cvxpy.reductions.dcp2cone
         - cvxpy.reductions.dcp2cone.atom_canonicalizers
         - cvxpy.reductions.eliminate_pwl
```

Comment:
@SteveDiamond @rileyjmurray 

Could you please double-check that we're still covering most of the public API here? I see a bunch of new modules have been added over time that aren't reflected here here (yet).

Also, the sole rename of `complex2real.{atom_canonicalizers -> canonicalizers}` seems inconsistent, given that all other `*.atom_canonicalizers` seem to remain as-is. I also left a [comment](https://github.com/cvxpy/cvxpy/pull/1976) on the respective upstream PR, feel free to take this conversation where you think it's best.

Reply:
Uh-oh. Didn't think to change the recipe. The rename of ``complex2real.atom_canonicalizers`` to ``complex2real.canonicalizers`` is because that code path includes canonicalizing ``Atom`` and ``Constraint`` objects. By contrast, ``dcp2cone.atom_canonicalizers`` indeed only handles canonicalization of ``Atom`` objects.

Technically, it seems that https://github.com/cvxpy/cvxpy/tree/master/cvxpy/reductions/dgp2dcp/atom_canonicalizers should be renamed to ``dgp2dcp.canonicalizers``, since it includes Constraint canonicalization. Let me look for others.

Reply:
It looks like renaming ``dgp2dcp.atom_canonicalizers`` to ``dgp2dcp.canonicalizers`` is the only thing we might want to do for consistency.

@SteveDiamond I think a bugfix to the 1.3.x release series is necessary to resolve the issue mentioned by @h-vetinari here https://github.com/conda-forge/cvxpy-feedstock/pull/75#issuecomment-1370469424. That being the case, we might also want to rename ``dgp2dcp.atom_canonicalizers`` to ``dgp2dcp.canonicalizers`` while we're at it.

Reply:
> That being the case, we might also want to rename `dgp2dcp.atom_canonicalizers` to `dgp2dcp.canonicalizers` while we're at it.

There's also `{eliminate_pwl, qp2quad_form}/atom_canonicalizers`...

Reply:
Those only process ``Atom`` objects, so the name is appropriate there. But I would be in favor of renaming all ``atom_canonicalizers`` folders into ``canonicalizers`` folders.

Reply:
> @SteveDiamond I think a bugfix to the 1.3.x release series is necessary to resolve the issue mentioned by @h-vetinari here [#75 (comment)](https://github.com/conda-forge/cvxpy-feedstock/pull/75#issuecomment-1370469424).

I have a PR too: https://github.com/cvxpy/cvxpy/pull/1998 🙃

Reply:
Separately, @h-vetinari, we're trying to adopt the convention that the public API consists exclusively of functionality accessible from the top-level CVXPY namespace. We moved a bunch of functionality to that namespace with 1.3. So if ``recipe/meta.yaml`` has semantics about the public API then we might want to make bigger changes here.

Reply:
> So if `recipe/meta.yaml` has semantics about the public API then we might want to make bigger changes here.

I don't know what you mean by "semantics"; the only thing it checks (well, in addition to running the test suite, which should be way more comprehensive anyway), is that the various `cvxpy.*` imports work. Since those are all in the toplevel namespace, the point seems to be moot anyway.

Reply:
So, in a way, the import tests here capture the _import_ API, and now show that it was changed (somewhat inconsistently) with 1.3.0 (if you unfold `meta.yaml` in the diff of this PR), you'll see what I mean.